velox/common/caching/SsdFile.cpp

```diff
@@ -438,6 +452,25 @@ void SsdFile::write(std::vector<CachePin>& pins) {
   }
 }
 
+bool SsdFile::write(
+    uint64_t offset,
+    uint64_t length,
+    const std::vector<iovec>& iovecs) {
+  const auto ret = folly::pwritev(fd_, iovecs.data(), iovecs.size(), offset);
+  if (ret != length) {
+    VELOX_SSD_CACHE_LOG(ERROR)
+        << "Failed to write to SSD, file name: " << fileName_ << ", fd: " << fd_
+        << ", size: " << iovecs.size() << ", offset: " << offset
+        << ", error code: " << errno
+        << ", error string: " << folly::errnoStr(errno);
+    ++stats_.writeSsdErrors;
+    // If write fails, we return without adding the pins to the cache. The
```

Comment:
Let's put this log in the caller which has this context?

Reply:
which means we need to write this multiple-line code twice? if that's the case maybe there's no need to have this new write function?